src/GraphQL/Types/DefaultGraphTypeFactory.cs

```diff
@@ -3,7 +3,7 @@ namespace GraphQL.Types;
 /// <summary>
 /// A generic factory to create instances of specific <typeparamref name="TGraphType"/> implementations from parameterless constructors.
 /// </summary>
-public class DefaultGraphTypeFactory<TGraphType> : IGraphTypeFactory<TGraphType> where TGraphType : IGraphType
+public class DefaultGraphTypeFactory<[RequireParameterlessConstructor] TGraphType> : IGraphTypeFactory<TGraphType> where TGraphType : IGraphType
```

Comment:
Should we just add a constraint?
```suggestion
public class DefaultGraphTypeFactory<TGraphType> : IGraphTypeFactory<TGraphType> where TGraphType : IGraphType, new()
```

Reply:
🤦😄

Reply:
Will also need to remove tests. There is a problem with testing internal attributes because I can't use them in tests. I have to test the API annotated with these attributes.